src/scripts/main.js

```diff
@@ -1,3 +1,32 @@
 'use strict';
 
-// write code here
+const listEl = document.querySelector('ul');
+
+const convertToNumber = str => Number(str.replace(/[^\d.-]/g, ''));
+
+const sortList = (list, sortParam) =>
+  [...list].sort(
+    (a, b) =>
+      convertToNumber(b.dataset[sortParam]) -
+      convertToNumber(a.dataset[sortParam]),
+  );
```

Comment:
The `sortList` function is doing a good job of sorting the list of elements based on a specific dataset attribute. However, it's not very flexible because it always sorts in descending order. Consider making this function more flexible by adding a parameter that specifies whether the sorting should be ascending or descending.